stdlib/asm/str/rlp.masm

```diff
@@ -0,0 +1,128 @@
+# Total Number of VM Cycles: 638 #
+# This is used for 5 field-elements -> 4 field-elements RLP converting. #
+# Used to decode a RLP code which contains up to 34 chars, the decoding result is ASCII #
+# input: [a, b_0, b_1, b_2, b_3, b_4] #
+#   a: a_hi -- whether the decoding result will be put into memory #
+#      a_lo -- only if a_hi is '1', this is the memory address to put the decoding result in  #
+#   b_0 to b_4 are the RLP code #
+# output: a, b, (C)
+#   a: the number of chars in the RLP Code #
+#   b: b_hi -- whether the decoding result will be put into memory #
+#      b_lo -- only if b_hi is '1', this is the memory address to put the decoding result in  #
+#   C: the decoding result if not stored into memory. If the decoding result is stored into memory, C doesn't exist on stack. #
+
+export.rlp_decode.2
+    push.adv.1
```

Comment:
I may be missing something here, but it seems like you read the inputs from the advice tape rather than assuming that they are on the stack already. I think the procedure should assume the 5 elements are already on the stack, and how these elements get there, should not be a concern of this procedure.